apps/backend/src/datasources/stfc/StfcFapDataSource.ts

```diff
@@ -0,0 +1,60 @@
+import { GraphQLError } from 'graphql';
+
+import { Fap } from '../../models/Fap';
+import { UserRoleShortCodeMap } from '../../models/User';
+import {
+  AssignChairOrSecretaryToFapInput,
+  AssignReviewersToFapArgs,
+} from '../../resolvers/mutations/AssignMembersToFapMutation';
+import { FapDataSource } from '../FapDataSource';
+import PostgresFapDataSource from '../postgres/FapDataSource';
+import { StfcUserDataSource } from './StfcUserDataSource';
+
+const stfcUserDataSource = new StfcUserDataSource();
+
+export default class StfcFapDataSource
+  extends PostgresFapDataSource
+  implements FapDataSource
+{
+  async assignChairOrSecretaryToFap(
+    args: AssignChairOrSecretaryToFapInput
+  ): Promise<Fap> {
+    const roles = await stfcUserDataSource.getUserRoles(args.userId);
+
+    if (
+      roles.find((role) => {
+        return (
+          role.shortCode === UserRoleShortCodeMap[args.roleId] ||
+          role.shortCode === 'user_officer'
+        );
+      })
+    ) {
+      return super.assignChairOrSecretaryToFap(args);
+    }
+
+    throw new GraphQLError(
+      `User does not have the correct role ${args.userId}`
+    );
+  }
+
+  async assignReviewersToFap(args: AssignReviewersToFapArgs): Promise<Fap> {
+    const members: number[] = [];
+
+    const usersWithRoles = await stfcUserDataSource.getUsersRoles(
+      args.memberIds
+    );
+
+    usersWithRoles.forEach((user) => {
+      !!user.roles.find(
+        (role) =>
+          role.shortCode === 'fap_reviewer' || role.shortCode === 'user_officer'
+      ) && members.push(user.userId);
+    });
+
+    if (members.length === args.memberIds.length) {
+      return super.assignReviewersToFap(args);
+    }
+
+    throw new GraphQLError('Some members did not have the correct Roles');
```

Comment:
The only info about users we have is there userIds at this point, and I wasn't too sure returning the IDs would be that useful, but I am happy to add them.

Reply:
I had thought it might make it easier for someone to resolve the problem my looking up the name and/or plugging those IDs into the panel member manager, but you know the use case and the systems better than I do so happy to leave it if you don't think it's worthwhile

Reply:
I you are right, and it might be worthwhile even if we have to make another call to the UOWS to get the information. I'll add it